### PR TITLE
Spack merge request

### PR DESCRIPTION
**Pull request by fangohr**
Realising feedback from Spack merge request review (https://github.com/spack/spack/pull/26933).

Forgot to explain adding of Debian package file in `Dockerfile`: this is necessary to be able to install the new clingo dependency resolver from binary packages in the container.

### CODE REVIEW

File: spack/package.py [hunk 1]
```diff
@@ -102,6 +101,10 @@ class Oommf(Package):
     depends_on("tcl", type=("build", "test", "run"))
     depends_on("xproto", type=("build"))
 
+    # Compilation with clang does not work yet (gcc works fine, nothing else tested)
+    # (https://github.com/spack/spack/pull/26933#pullrequestreview-789754233)
+    conflicts("%clang")
```
Comment (lang-m) [inline spack/package.py:106]: OOMMF should have some clang support: https://github.com/fangohr/oommf/blob/master/oommf/config/platforms/clang-support.tcl
Maybe some changes are required somewhere in here?: https://github.com/fangohr/oommf/blob/67fa0d69eadbbb9eef320babd07910f6d7b4e089/oommf/config/platforms/linux-x86_64.tcl#L100
I think exporting the correct `OOMMF_CPP` environment variable could be sufficient: https://github.com/fangohr/oommf/blob/67fa0d69eadbbb9eef320babd07910f6d7b4e089/oommf/config/platforms/linux-x86_64.tcl#L43 (this takes precedence over L100 above)

File: spack/package.py [hunk 2]
```diff
@@ -135,6 +140,25 @@ def get_oommf_path(self, prefix):
...
+    @property
+    def test_env(self):
+        """Create environment in which post-install tests can be run."""
+        # Make sure the correct OOMMF config.tcl is found.
+        # This environment variable (OOMMF_ROOT) seems not to be
+        # set at this point, so we have to set it manually for the test:
+        oommfdir = self.get_oommf_path(self.prefix)
+        test_env_ = {}
+        test_env_["OOMMF_ROOT"] = oommfdir
+        return test_env_
```
Comment (lang-m) [inline spack/package.py:160]: This could be a single line without and additional variable.

File: spack/package.py [hunk 3]
```diff
@@ -167,10 +191,6 @@ def install(self, spec, prefix):
             for f in install_files:
                 install(os.path.join(oommfdir, f), prefix.bin)
 
-    def setup_build_environment(self, env):
-        # Seems we don't need to do anything here
-        pass
```
Comment (lang-m) [inline spack/package.py:172]: Maybe we can specify `OOMMF_CPP` in here?

File: spack/package.py [hunk 4]
```diff
@@ -181,85 +201,53 @@ def setup_run_environment(self, env):
...
-        oommf_tcl_path = join_path(spec.prefix.bin, "oommf.tcl")
-        # put the command together and execute
-        output = tclsh(
-            oommf_tcl_path, "+version", output=str.split, error=str.split, env=test_env
+        output = self.tclsh(
+            self.oommf_tcl_path,
+            "+version",
+            output=str.split,
+            error=str.split,
+            env=test_env,
```
Comment (lang-m) [inline spack/package.py:212]: Could be `self.test_env` allowing us to remove L204.

File: spack/package.py [hunk 5]
```diff
@@ -181,85 +201,53 @@ def setup_run_environment(self, env):
...
-        oommf_tcl_path = join_path(spec.prefix.bin, "oommf.tcl")
-        # put the command together and execute
-        output = tclsh(
-            oommf_tcl_path, "+platform", output=str.split, error=str.split, env=test_env
+        output = self.tclsh(
+            self.oommf_tcl_path,
+            "+platform",
+            output=str.split,
+            error=str.split,
+            env=test_env,
```
Comment (lang-m) [inline spack/package.py:231]: `self.test_env` (and removing L219)

File: spack/package.py [hunk 6]
```diff
@@ -181,85 +201,53 @@ def setup_run_environment(self, env):
 
     @run_after("install")
     def check_install_version(self):
```
Comment (lang-m) [inline spack/package.py:203]: This function is almost the same as `check_install_platform` and `check_install_stdprob3`. An additional helper function `_check_install(*oommf_args)` could avoid some code duplication.

**Review by lang-m — COMMENTED** [review]
Two comment to `test` (lines 257 onwards) [GitHub doesn't allow me to add comments in there]:
- Can we use any of the new properties in lines 261-263?
- The output of `oommf +version` has changed in 2.0a3 (`oommf.tcl` has been replaced by `OOMMF`).

**Comment by fangohr** [comment]
> Two comment to `test` (lines 257 onwards) [GitHub doesn't allow me to add comments in there]:
>
> * Can we use any of the new properties in lines 261-263?
I thought about this too, but I prefer to keep these separate: the `test` function is testing the installed package (so presumably the environment settings are the same as if you use `spack load oommf`. In any case, the environment settings here are different from the post-install tests which use the properties. So I don't want to mix them up, and prefer the `test` to be self-contained.

**Comment by fangohr** [comment]
> Two comment to `test` (lines 257 onwards) [GitHub doesn't allow me to add comments in there]:
> 
> * The output of `oommf +version` has changed in 2.0a3 (`oommf.tcl` has been replaced by `OOMMF`).

Oohh - well spotted. The test didn't fail because we had plenty of other `oommf.tcl` in the output. 

As I don't know how to make this version dependent, I am now checking for the `info:` string from:
```
$> oommf.tcl +version
<2145466> OOMMF 2.0a3  info:
OOMMF 2.0a3
```

File: spack/package.py (hunk 1 above)
Comment (fangohr) [reply spack/package.py:106]: Nice. I'll create a new issue for this to get to some closure with the MR and gcc soon.

File: spack/package.py (hunk 3 above)
Comment (fangohr) [reply spack/package.py:172]: Good idea. Mentioned in issue #31 .

File: spack/package.py (hunk 6 above)
Comment (fangohr) [reply spack/package.py]: Good idea. That's done now.

**Pull request merged**